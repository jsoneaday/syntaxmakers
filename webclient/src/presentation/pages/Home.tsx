import developer from "../../presentation/theme/assets/programmer.png";
import employer from "../../presentation/theme/assets/businessman.png";
import "../../presentation/theme/home.css";
import { Link } from "react-router-dom";

export default function Home() {
  return (
    <div className="home">
      <div className="panel-col home-item-dev">
        <Link to="developer" className="home-item-link">
          <img className="home-icon" src={developer} />
          <div className="title-font home-item-content">I'm a Developer</div>
        </Link>
      </div>

      <div className="panel-col home-item-emp">
        <Link to="employer" className="home-item-link">
          <img className="home-icon" src={employer} />
          <div className="title-font home-item-content">I'm an Employer</div>
        </Link>
      </div>
    </div>
  );
}
