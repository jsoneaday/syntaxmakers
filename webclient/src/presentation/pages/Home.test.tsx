import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "./Home";
import { RouterProvider, createBrowserRouter } from "react-router-dom";

const router = createBrowserRouter([
  {
    path: "/",
    element: <Home />,
  },
]);

function WrappedHome() {
  return <RouterProvider router={router} />;
}

describe("Test Home", () => {
  it("check snapshot", async () => {
    render(<WrappedHome />);
    const home = screen.getByTestId("home-page");

    expect(home).toBeInTheDocument();
    expect(home).toMatchSnapshot();
  });
});
