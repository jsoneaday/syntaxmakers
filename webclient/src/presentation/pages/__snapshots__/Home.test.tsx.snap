// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home matches snapshot 1`] = `
<div
  class="home"
  data-testid="home-page"
>
  <div
    class="panel-col home-item-dev"
  >
    <a
      class="home-item-link"
      data-testid="dev-link"
      href="/developer"
    >
      <img
        class="home-icon"
        src="default"
      />
      <div
        class="title-font home-item-content"
      >
        I'm a Developer
      </div>
    </a>
  </div>
  <div
    class="panel-col home-item-emp"
  >
    <a
      class="home-item-link"
      data-testid="emp-link"
      href="/employer"
    >
      <img
        class="home-icon"
        src="default"
      />
      <div
        class="title-font home-item-content"
      >
        I'm an Employer
      </div>
    </a>
  </div>
</div>
`;
